specs/agents/metadata.md

```diff
@@ -136,6 +136,37 @@ For official Elastic agents, the agent name should just be the name of the langu
 
 Services running on AWS Lambda [require specific values](tracing-instrumentation-aws-lambda.md) for some of the above mentioned fields.
 
+#### Installation method
+
+Most of the APM Agents can be installed in several ways. Agents SHOULD collect information about the used installation method and send it in the `service.agent.installation.method` field within the metadata.
+
+The intention of this field is to drive telemetry so there is a way to know which installation methods are commonly used. This field MUST produce data with very low cardinality, therefore agents SHOULD use one of the values defined below.
+
+There are some well-known installation methods which can be used by multiple agents. In those cases, agents SHOULD send the following values in `service.agent.installation.method`:
+
+- `aws-lambda-layer`: when the agent runs in AWS lambda and uses [the AWS lambda extension](https://github.com/elastic/apm-aws-lambda).
```

Comment:
Good point - I didn’t know about that.

My thinking was this: we did a bunch of work in most agents to support AWS lambda and the outcome of that work was an “agent installation” which enables proper lambda support. All those rely on our AWS lambda extension, right? 

My intention with this part was that we want to know if the user uses this installation or not.

> Node.js and Java have a layer available for adding the agent to a lambda function while the Python agent does not.

I think we should then adapt the value we send. In this case it’s probably better to go with `aws-lambda` (and drop the `layer` in the value).

> are we interested in collecting anything else about the way the agent is added to a lambda function?

We only have 1 field https://github.com/elastic/apm-server/issues/9622 so I don’t think we have the option to collect more - unless we rethink the new field.

Overall I feel the added value is to know if the installation uses our things which we specifically developed to support AWS support - whether an agent itself uses an AWS layer or not is not relevant for telemetry.

Theoretically we could split this into multiple values (e.g. have `aws-lambda` and `aws-lambda-layer`), but that increases the cardinality with very little added value.

What do you think about this?

Reply:
Hi @basepi and @beniwohli 
We talked about this in the Node.js agent meeting today. The Node.js agent and the Java agent can both be added via a lambda layer. We thought that would be the interesting differentiator in telemetry data (and not whether the lambda extension is used).
For the python agent, what would be interesting to know when the agent is used in a Lambda environment? We thought some candidates would be:
- that the function was instrumented
- that the lambda extension was used
- that the python agent was installed via a layer. We know this isn't possible now but we guessed that the wrapper script work might allow that to be a possible offering in the future.

Reply:
We only get data if the function was instrumented, so that's probably not an important data point, but the latter two are definitely relevant (I expect a layer will be possible once we have the wrapper script)

Reply:
> that the lambda extension was used

Is there a way to know this and set `agent.installation.method`?

Reply:
I figured we would do it based on `ELASTIC_APM_LAMBDA_APM_SERVER` being set, and/or `ELASTIC_APM_SERVER_URL` being localhost.

Reply:
Thanks @basepi 

So, how do we proceed? 

Options:
1) We name it `aws-lambda` across all agents - some will do the detection based on the layer, Python (and maybe some others) will do it currently based on environment variables. The info we get out here is this: we'll know that the agent is installed based on the ["Monitoring AWS Lambda [X] funtions"](https://www.elastic.co/guide/en/apm/agent/python/current/lambda-support.html) part of the given agent's doc.
2) We leave it with `aws-lambda-layer` and Python does not report this currently - once installation via layer based on wrapper script is released, Python can start reporting this.
3) Some other ideas?

I think option 1 is the most value - I think reporting this for Python is still useful.

Reply:
For Lambda, I see a few useful data points for python:

1. Is it lambda? (We already know this from metadata)
2. Is it lambda with the extension? (We can collect environment variables)
3. Is it lambda via the lambda layer? (Maybe we set a variable internally or an environment variable to differentiate from 2?)

I'm torn on whether 2 and 3 are interchangeable and whether they should be lumped together. We did have someone using the `container image` install method of the extension the other day. That would be a version of 2, and data which would be lost if we lump 2 and 3 together.

But your `aws-lambda` proposal is still an improvement on the data we have today, so I don't feel super strongly.

Reply:
So if we're focused on install method, should we start a proposal for another field for invocation/activation method? For python specifically, that's more interesting to us for everything that's not lambda. Most of the time the python agent will be installed via `pip`, but whether it's invoked via code changes or the wrapper script is definitely telemetry we'd like to know for the python agent.

Reply:
> Keeping the original idea: let's focus on "installation method".

I think what we mean with that is that we don't want to mix up how an agent is used on AWS Lambda with the fact whether it's used together with the Lambda extension or not.

So renaming it to `agent.activation.method` still makes sense.